app/src/main/java/io/appium/uiautomator2/model/settings/AbstractSetting.java

```diff
@@ -22,35 +22,45 @@
 public abstract class AbstractSetting<T> implements ISetting {
 
     private final Class<T> valueType;
+    private final String settingName;
 
-    public AbstractSetting(Class<T> valueType) {
+    public AbstractSetting(Class<T> valueType, String settingName) {
         this.valueType = valueType;
+        this.settingName = settingName;
     }
 
-    public void updateSetting(Object value) {
-        Logger.debug(String.format("Set the %s to %s", getSettingName(), String.valueOf(value)));
+    public void update(Object value) {
+        Logger.debug(String.format("Set the %s to %s", getName(), String.valueOf(value)));
         T convertedValue = convertValue(value);
         try {
             apply(convertedValue);
         } catch (Exception e) {
-            Logger.error(String.format("Unable to update the setting %s: %s", getSettingName(), e.toString()));
+            Logger.error(String.format("Unable to update the setting %s: %s", getName(), e.toString()));
         }
     }
 
-    public abstract String getSettingName();
+    public String getName() {
+        return settingName;
+    };
 
     public Class<T> getValueType() {
         return valueType;
     }
 
+    public abstract T getValue();
+
     protected abstract void apply(T value);
 
     private T convertValue(Object value) {
-        if (!valueType.isInstance(value)) {
+        try {
+            if (valueType == Long.class && value instanceof Number) {
```

Comment:
I assume it is obvious than if the first condition here is true then the second one is true as well.

Reply:
Can you explain your point in more detail? I can't find any correlation between these two conditions.

Reply:
I mean any value of Long class is a number instance

Reply:
```valueType``` represents the type of setting, but ```value``` is the desired value. Therefore these variables can belong to different classes.